aten/src/ATen/native/cpu/Loops.h

```diff
@@ -186,7 +186,9 @@ static inline void unroll_contiguous_scalar_checks(
 template <typename func_t>
 void cpu_kernel(TensorIterator& iter, func_t&& op) {
   using traits = function_traits<func_t>;
-  TORCH_INTERNAL_ASSERT(iter.ntensors() >= traits::arity + 1);
+  // this could be extended to work with void return types
+  TORCH_INTERNAL_ASSERT(iter.ninputs() == traits::arity);
+  TORCH_INTERNAL_ASSERT(iter.noutputs() == 1);
```

Comment:
This is true for current kernels, but in general we should not hardcode number of outputs to 1? E.g. divmod will need 2 outputs, some activation backward functions already need 2 outputs, and nvidia has a PR out that enables multiple outputs for gpu tensorIterator kernels https://github.com/pytorch/pytorch/pull/37969/files.

Reply:
thanks for pointing this out, I wasn't aware of that PR.

I'm happy to extend any of these if we need, I just wanted to start with the most restrictive.  So I can change the comments from "// this could be extended to work with void return types" to "// this could be extended to work with void or multiple (tuple) return types"